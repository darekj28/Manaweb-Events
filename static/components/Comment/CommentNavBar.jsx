var React = require('react');
var Link = require('react-router').Link;
var browserHistory = require('react-router').browserHistory;
import NotificationsDropdown from "../GenericNavBar/NotificationsDropdown.jsx";
import AccountDropdown from "../GenericNavBar/AccountDropdown.jsx";

export default class CommentNavBar extends React.Component {
	handleSearch() {
		this.props.onSearch(this.searchText.value);
	}
	componentDidMount() {
		var searchVisible = 0;
		// $('.navbar-search-form').hide();
		$('.navbar-search-form').show();

		// $('[data-toggle="search"]').click(function(){
	 //        if(searchVisible == 0){
	 //            searchVisible = 1;
	 //            $(this).parent().addClass('active');
	 //            $('.navbar-search-form').fadeIn(function(){
	 //                $('.navbar-search-form input').focus();
	 //            });
	 //        } else {
	 //            searchVisible = 0;
	 //            $(this).parent().removeClass('active');
	 //            $(this).blur();
	 //            $('.navbar-search-form').fadeOut(function(){
	 //                $('.navbar-search-form input').blur();
	 //            });
	 //        } 
	 //    });
	    $('#searchInput').keypress(function(event) {
	    	if(event.keyCode == 13){ 
			   event.preventDefault();
			}
	    });
	    $('.SearchNavBarGlyphicon').focus(function() {
	    	$(this).blur();
	    });
	}
	render() {
		return (
			<nav className="navbar navbar-default" role="navigation">
				<div className="container">
					<div className="navbar-header">
						<button type="button" className="SearchNavBarGlyphicon navbar-toggle" data-toggle="collapse" 
										data-target="#bs-example-navbar-collapse-1">
				            <span className="sr-only">Toggle navigation</span>
				            <span className="icon-bar"></span>
				            <span className="icon-bar"></span>
				            <span className="icon-bar"></span>
				          </button>
				          <Link onClick={browserHistory.goBack} className="SearchNavBarGlyphicon navbar-brand navbar-brand-logo">
				                <span className="glyphicon glyphicon-chevron-left"></span>
				              </Link>
				        </div>
				        <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
				          <ul className="nav navbar-nav navbar-right">
		                    {/* <li className="">
		                        <a href="javascript:void(0)" data-toggle="search">
		 							<span className="glyphicon glyphicon-search"></span>
		                        </a>
		                    </li> */}
 				          	<NotificationsDropdown/>
				          	<AccountDropdown currentUser = {this.props.currentUser} name={this.props.name}/>
				          </ul>
				         <form className="navbar-form navbar-right navbar-search-form" role="search">                  
			                 <div className="form-group">
			                 	  <div className="input-group input-group-unstyled">
				                      <input type="text" value={this.props.searchText} ref={(input) => this.searchText = input} 
				                      			id="searchInput" className="form-control" placeholder="Search..." 
				                      			onChange={this.handleSearch.bind(this)}/>
								  </div>		  
			                 </div> 
			              </form>
				        </div>
				      </div>
				    </nav>
		)
	}
}