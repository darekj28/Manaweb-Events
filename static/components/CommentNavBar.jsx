var React = require('react');
var Link = require('react-router').Link;
import Notifications from "Notifications.jsx";
import AccountDropdown from "AccountDropdown.jsx";

export default class SearchNavBar extends React.Component {
	constructor(props) {
		super(props);
		this.handleSearch = this.handleSearch.bind(this);
	}
	componentDidMount() {
		var searchVisible = 0;
		$('.navbar-search-form').hide();
		$('[data-toggle="search"]').click(function(){
	        if(searchVisible == 0){
	            searchVisible = 1;
	            $(this).parent().addClass('active');
	            $('.navbar-search-form').fadeIn(function(){
	                $('.navbar-search-form input').focus();
	            });
	        } else {
	            searchVisible = 0;
	            $(this).parent().removeClass('active');
	            $(this).blur();
	            $('.navbar-search-form').fadeOut(function(){
	                $('.navbar-search-form input').blur();
	            });
	        } 
	    });
	    $('#searchInput').keypress(function(event) {
	    	if(event.keyCode == 13){ 
			   event.preventDefault();
			}
	    });
	    $('#home').focus(function() {
	    	$(this).blur();
	    });
	}
	handleSearch() {
		this.props.onSearch(this.searchText.value);
	}
	render() {
		var that = this;
		return (
			<nav className="navbar navbar-default" role="navigation">
				<div className="container">
					<div className="navbar-header">
						<button type="button" className="navbar-toggle" data-toggle="collapse" 
										data-target="#bs-example-navbar-collapse-1">
				            <span className="sr-only">Toggle navigation</span>
				            <span className="icon-bar"></span>
				            <span className="icon-bar"></span>
				            <span className="icon-bar"></span>
				          </button>
				          <Link to="/" className="navbar-brand navbar-brand-logo">
				                <span className="glyphicon glyphicon-chevron-left"></span>
				              </Link>
				        </div>
				        <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
				          <ul className="nav navbar-nav navbar-right">
		                    <li className="">
		                        <a href="javascript:void(0)" data-toggle="search">
		 							<span className="glyphicon glyphicon-search"></span>
		                        </a>
		                    </li>
				          	<Notifications/>
				          	<AccountDropdown/>
				          </ul>
				         <form className="navbar-form navbar-right navbar-search-form" role="search">                  
			                 <div className="form-group">
			                 	  <div className="input-group input-group-unstyled">
				                      <input type="text" value={this.props.searchText} ref={(input) => this.searchText = input} 
				                      			id="searchInput" className="form-control" placeholder="Search..." 
				                      			onChange={this.handleSearch}/>
				                      <div className = "input-group-addon"></div>
								  </div>
								  
			                 </div> 
			              </form>
				        </div>
				      </div>
				    </nav>
		)
	}
}