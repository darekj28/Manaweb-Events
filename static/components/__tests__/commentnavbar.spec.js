import CommentNavBar from '../CommentNavBar.jsx';

describe("CommentNavBar", () => {
	it("renders correctly", () => {
		const cnb = renderer.create(
			<CommentNavBar searchText="a" onSearch={jest.fn()} name="Jay Kim"/>
			);
		expect(cnb).toMatchSnapshot();
	});
	it("should call onSearch with text in textbox when user types", () => {
		expect(true).toBe(true);
	});
});