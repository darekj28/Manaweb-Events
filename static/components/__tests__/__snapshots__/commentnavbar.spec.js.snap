exports[`CommentNavBar renders correctly 1`] = `
<nav
  className="navbar navbar-default"
  role="navigation">
  <div
    className="container">
    <div
      className="navbar-header">
      <button
        className="navbar-toggle"
        data-target="#bs-example-navbar-collapse-1"
        data-toggle="collapse"
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
      <a
        className="navbar-brand navbar-brand-logo"
        onClick={[Function]}
        style={Object {}}>
        <span
          className="glyphicon glyphicon-chevron-left" />
      </a>
    </div>
    <div
      className="collapse navbar-collapse"
      id="bs-example-navbar-collapse-1">
      <ul
        className="nav navbar-nav navbar-right">
        <li
          className="">
          <a
            data-toggle="search"
            href="javascript:void(0)">
            <span
              className="glyphicon glyphicon-search" />
          </a>
        </li>
        <li
          className="dropdown">
          <a
            className="dropdown-toggle"
            data-toggle="dropdown"
            href="#">
            <span
              className="glyphicon glyphicon-envelope" />
          </a>
          <ul
            className="dropdown-menu">
            <li>
              <a
                href="#">
                Insert notifications here.
              </a>
            </li>
            <li>
              <a
                href="#">
                Insert notifications here.
              </a>
            </li>
            <li>
              <a
                href="#">
                Insert notifications here.
              </a>
            </li>
          </ul>
        </li>
        <li
          className="dropdown">
          <a
            className="dropdown-toggle"
            data-toggle="dropdown"
            href="#">
            <span
              className="glyphicon glyphicon-user" />
          </a>
          <ul
            className="dropdown-menu">
            <li>
              <a
                href="#">
                Jay Kim
              </a>
            </li>
            <li
              className="divider" />
            <li>
              <a
                href="/settings">
                Settings
              </a>
            </li>
            <li>
              <a
                href="#">
                Your Posts
              </a>
            </li>
            <li>
              <a
                href="/logout">
                Logout
              </a>
            </li>
          </ul>
        </li>
      </ul>
      <form
        className="navbar-form navbar-right navbar-search-form"
        role="search">
        <div
          className="form-group">
          <div
            className="input-group input-group-unstyled">
            <input
              className="form-control"
              id="searchInput"
              onChange={[Function]}
              placeholder="Search..."
              type="text"
              value="a" />
            <div
              className="input-group-addon" />
          </div>
        </div>
      </form>
    </div>
  </div>
</nav>
`;
